crates/ide_completion/src/context.rs

```diff
@@ -337,25 +337,25 @@ impl<'a> CompletionContext<'a> {
                     },
                     ast::RecordExprFieldList(_it) => {
                         cov_mark::hit!(expected_type_struct_field_without_leading_char);
-                        self.token.prev_sibling_or_token()
-                            .and_then(|se| se.into_node())
-                            .and_then(|node| ast::RecordExprField::cast(node))
-                            .and_then(|rf| self.sema.resolve_record_field(&rf).zip(Some(rf)))
-                            .map(|(f, rf)|(
-                                Some(f.0.ty(self.db)),
-                                rf.field_name().map(NameOrNameRef::NameRef),
+                        // wouldn't try {} be nice...
+                        (|| {
+                            let record_ty = self.sema.type_of_expr(&ast::Expr::cast(node.parent()?)?)?;
+                            let expr_field = self.token.prev_sibling_or_token()?
+                            .into_node()
+                                      .and_then(|node| ast::RecordExprField::cast(node))?;
+                            let field = self.sema.resolve_record_field(&expr_field)?.0;
+                            Some((
+                                record_ty.field_type(self.db, field),
```

Comment:
Do we really need field type here? Perhaps its better if `self.sema.resolve_record_field` returns a `Type` alongside the Field?

Reply:
I think `field_type` is a good method to have anyway :thinking:

Reply:
Yeah, this is what I am unsure about... It does seems like a neat API for "how do I project type through the filed", but I wonder if we can get by without that functionality at all. 

My thinking initially was that `.ty` on a `Field` wold need some kind of `subst` to be useful. But the `field_ty` API feels better in a sense that, instead of managing subst, you can project a type with the correct substs. But then, perhaps one can always just get the correct type instead?

I feel that historically "record" approaches worked better for us than "recompute". Projecting the field is "recompute", and just getting the type from the syntax node is "record".

Do you have other use-cases in mind for `.field_type`?

Reply:
Hmm you're probably right.